cupy/lib/stride_tricks.py

```diff
@@ -37,3 +38,117 @@ def as_strided(x, shape=None, strides=None):
+
+def sliding_window_view(x, window_shape, axis=None, *,
+                        subok=False, writeable=False):
+    """
+    Create a sliding window view into the array with the given window shape.
+
+    Also known as rolling or moving window, the window slides across all
+    dimensions of the array and extracts subsets of the array at all window
+    positions.
+
+
+	Parameters
+    ----------
+    x : array_like
+        Array to create the sliding window view from.
+    window_shape : int or tuple of int
+        Size of window over each axis that takes part in the sliding window.
+        If `axis` is not present, must have same length as the number of input
+        array dimensions. Single integers `i` are treated as if they were the
+        tuple `(i,)`.
+    axis : int or tuple of int, optional
+        Axis or axes along which the sliding window is applied.
+        By default, the sliding window is applied to all axes and
+        `window_shape[i]` will refer to axis `i` of `x`.
+        If `axis` is given as a `tuple of int`, `window_shape[i]` will refer to
+        the axis `axis[i]` of `x`.
+        Single integers `i` are treated as if they were the tuple `(i,)`.
+    subok : bool, optional
+        If True, sub-classes will be passed-through, otherwise the returned
+        array will be forced to be a base-class array (default).
+    writeable : bool, optional
+        When true, allow writing to the returned view. The default is false,
+        as this should be used with caution: the returned view contains the
+        same memory location multiple times, so writing to one location will
+        cause others to change.
+
+    Returns
+    -------
+    view : ndarray
+        Sliding window view of the array. The sliding window dimensions are
+        inserted at the end, and the original dimensions are trimmed as
+        required by the size of the sliding window.
+        That is, ``view.shape = x_shape_trimmed + window_shape``, where
+        ``x_shape_trimmed`` is ``x.shape`` with every entry reduced by one less
+        than the corresponding window size.
+
+
+    See also
+    --------
+    numpy.lib.stride_tricks.as_strided
+
+	Notes
+	--------
+	This function is adapted from numpy.lib.stride_tricks.as_strided.
+
+	Examples
+    --------
+    >>> x = _cupy.arange(6)
+    >>> x.shape
+    (6,)
+    >>> v = sliding_window_view(x, 3)
+    >>> v.shape
+    (4, 3)
+    >>> v
+    array([[0, 1, 2],
+           [1, 2, 3],
+           [2, 3, 4],
+           [3, 4, 5]])
+
+    """
+    window_shape = (tuple(window_shape)
+                    if np.iterable(window_shape)
+                    else (window_shape,))
+
+    # -- because normalize_axis_tuple does not handle lists:
+    # -- with Xarray it causes error : 
+    # -- TypeError: 'list' object cannot be interpreted as an integer
+    axis = (
+        axis[0]
+        if np.iterable(axis) and len(axis) == 1
+        else axis)
```

Comment:
I think we should fix this in xarray.

Though perhaps cupy should just vendor the upstream `normalize_axis_tuple` which handles lists.
```
np.core.numeric.normalize_axis_tuple([1, 2, -1], 4) # (1, 2, 3)
```

Reply:
Thanks and done! An issue regarding this is also filed with Xarray :  https://github.com/pydata/xarray/issues/7890